libz-rs-sys/src/gz.rs

```diff
@@ -2269,6 +2269,64 @@ pub unsafe extern "C-unwind" fn gzgets(file: gzFile, buf: *mut c_char, len: c_in
     }
 }
 
+/// Dynamically update the compression level and strategy for `file`. See the
+/// description of [`deflateInit2`] for the meaning of these parameters. Previously
+/// provided data is flushed before applying the parameter changes.
+///
+/// # Returns
+///
+/// - `Z_OK` on success.
+/// - `Z_STREAM_ERROR` if the file was not opened for writing.
+/// - `Z_ERRNO` if there is an error writing the flushed data.
+/// - `Z_MEM_ERROR` if there is a memory allocation error.
+///
+/// # Safety
+///
+/// - `file`, if non-null, must be an open file handle obtained from [`gzopen`] or [`gzdopen`].
+#[cfg_attr(feature = "export-symbols", export_name = crate::prefix!(gzsetparams))]
+pub unsafe extern "C-unwind" fn gzsetparams(file: gzFile, level: c_int, strategy: c_int) -> c_int {
+    let Ok(strategy) = Strategy::try_from(strategy) else {
+        return Z_STREAM_ERROR;
+    };
+    let Some(state) = (unsafe { file.cast::<GzState>().as_mut() }) else {
+        return Z_STREAM_ERROR;
+    };
+
+    // Check that we're writing and that there's no error.
+    if state.mode != GzMode::GZ_WRITE || state.err != Z_OK || state.direct {
+        return Z_STREAM_ERROR;
+    }
+
+    // If no change is requested, then do nothing.
+    if level == state.level as c_int && strategy == state.strategy {
```

Comment:
this can be `c_int::from(state.level)` (communicates that no truncation can happen)